### PR TITLE
Fix different processing of json values offline and online

### PR DESCRIPTION
**Pull request by guperrot**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Fixes for #60954 and #60953 where json was inconsistent between offline and online scenarios.
The json value must always be customer content in sqlite and in memory objects.

### CODE REVIEW

**Comment by guperrot** [comment]
Delete online broken. It's also indirectly related to remaining unit test failures.

```
2019-04-29 18:49:04.714641-0700 SasquatchPuppet[7420:1727737] [AppCenterData] ERROR: +[MSDocumentUtils documentWrapperFromDocumentData:documentType:eTag:lastUpdatedDate:partition:documentId:pendingOperation:fromDeviceCache:]/99 Error deserializing data: The data couldn’t be read because it isn’t in the correct format.
```

**Comment by guperrot** [comment]
> Error deserializing data

That's on develop, nevermind: AB#61035

File: AppCenterData/AppCenterData/Internal/Util/MSDocumentUtils.m [hunk 1]
```diff
@@ -120,7 +120,8 @@ + (MSDocumentWrapper *)documentWrapperFromDictionary:(NSObject *)object
   if (![MSDocumentUtils isReferenceDictionaryWithKey:object key:kMSDocumentIdKey keyType:[NSString class]] ||
       ![MSDocumentUtils isReferenceDictionaryWithKey:object key:kMSDocumentTimestampKey keyType:[NSNumber class]] ||
       ![MSDocumentUtils isReferenceDictionaryWithKey:object key:kMSDocumentEtagKey keyType:[NSString class]] ||
-      ![MSDocumentUtils isReferenceDictionaryWithKey:object key:kMSPartitionKey keyType:[NSString class]]) {
+      ![MSDocumentUtils isReferenceDictionaryWithKey:object key:kMSPartitionKey keyType:[NSString class]] ||
+      ![MSDocumentUtils isReferenceDictionaryWithKey:object key:kMSDocumentKey keyType:[NSDictionary class]]) {
```
Comment (guperrot) [inline AppCenterData/AppCenterData/Internal/Util/MSDocumentUtils.m]: That's not correct as delete has null customer payload.

Comment (guperrot) [reply AppCenterData/AppCenterData/Internal/Util/MSDocumentUtils.m]: And this is what makes unit test fails.

**Comment by guperrot** [comment]
Screen Shot 2019-04-29 at 7 52 14 PM

The test is actually right. On develop, even on error we get etag, partition and date being set.
This PR introduces a regression where we don't set these fields anymore because we fail early, the return error statements were actually not a good change @jwallra @achocron .

File: AppCenterData/AppCenterData/Internal/Util/MSDocumentUtils.m (hunk 1 above)
Comment (guperrot) [reply AppCenterData/AppCenterData/Internal/Util/MSDocumentUtils.m]: Fixed in https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1561/commits/712173afaa8b24c169608caacbc00d922eae9a5a

**Comment by guperrot** [comment]
@jwallra @achocron the remaining tests are failing for a good reason.
We return error too early now in utils, and thus some metadata is missing.
We need to keep populating those metadata on JSON error (so keep the asserts as is and fix utils code).

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m [hunk 2]
```diff
@@ -73,7 +73,7 @@ - (void)testReadAppDocumentFromLocalDatabase {
   // If
   NSString *documentId = @"12829";
   NSString *eTag = @"398";
-  NSString *jsonString = @"{ \"document\": {\"key\": \"value\"}}";
```
Comment (YiMysty) [inline AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: Is a specific reason to change the test string here? :)

Comment (achocron) [reply AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: Yes, now we do not store the root json string in storage, just the json of the document property

File: AppCenterData/AppCenterDataTests/MSDocumentUtilsTests.m [hunk 3]
```diff
@@ -152,7 +111,7 @@ - (void)testDocumentWrapperFromDictionaryWithDocument {
   XCTAssertTrue([[document eTag] isEqualToString:@"etag"]);
   XCTAssertNotNil([document lastUpdatedDate]);
   XCTAssertTrue([[document partition] isEqualToString:@"readonly"]);
-  XCTAssertNotNil([document jsonValue]);
+  XCTAssertNil([document jsonValue]);
```
Comment (jaelim-ms) [inline AppCenterData/AppCenterDataTests/MSDocumentUtilsTests.m]: Why should it be `nil`? Was the test wrong before?

Comment (achocron) [reply AppCenterData/AppCenterDataTests/MSDocumentUtilsTests.m]: Yeah now there is no metadata in the json value, just the document itself. It was inconsistent before.

Comment (guperrot) [reply AppCenterData/AppCenterDataTests/MSDocumentUtilsTests.m]: This is nil on document parsing error now. Before you could have deserialized value non nil and failed jsonValue being nil.

**Pull request merged**